### PR TITLE
chore: geo category lint fixes for cdk v2

### PR DESCRIPTION
**Pull request by akshbhu**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-geo/src/service-stacks/geofenceCollectionStack.ts [hunk 1]
```diff
@@ -100,43 +108,45 @@ export class GeofenceCollectionStack extends BaseStack {
...
+  private constructCollectionPolicyResources(collectionResource: cdk.CustomResource): void {
     Object.keys(this.groupPermissions).forEach((group: string) => {
       // This is a work-around until the CollectionArn is included in the `UpdateGeofenceCollection` output
       const outputGeofenceCollectionARN = cdk.Fn.sub('arn:aws:geo:${region}:${account}:geofence-collection/${collectionName}', {
         region: this.geofenceCollectionRegion,
         account: cdk.Fn.ref('AWS::AccountId'),
-        collectionName: collectionResource.getAtt('CollectionName').toString()
+        collectionName: collectionResource.getAtt('CollectionName').toString(),
       });
+      // eslint-disable-next-line max-len
```
Comment (sobolk) [inline packages/amplify-category-geo/src/service-stacks/geofenceCollectionStack.ts:127]: should we rather add some `\n` to break it down ?

**Review by sobolk — DISMISSED** [review]

File: packages/amplify-category-geo/src/service-utils/mapUtils.ts [hunk 2]
```diff
@@ -28,7 +34,8 @@ export const createMapResource = async (context: $TSContext, parameters: MapPara
   const mapStack = new MapStack(new App(), 'MapStack', { ...parameters, ...templateMappings, authResourceName });
   generateTemplateFile(mapStack, parameters.name);
   saveCFNParameters(parameters);
-  stateManager.setResourceInputsJson(pathManager.findProjectRoot(), category, parameters.name, { groupPermissions: parameters.groupPermissions });
+  stateManager.setResourceInputsJson(pathManager.findProjectRoot(),
+    category, parameters.name, { groupPermissions: parameters.groupPermissions });
```
Comment (jhockett) [inline packages/amplify-category-geo/src/service-utils/mapUtils.ts:38]: nit: formatting and pathManager.findProjectRoot() gets called anyways if undefined is passed into the first arrgument

```suggestion
  stateManager.setResourceInputsJson(
    undefined,
    category,
    parameters.name,
    { groupPermissions: parameters.groupPermissions },
  );
```

File: packages/amplify-category-geo/src/service-utils/mapUtils.ts [hunk 3]
```diff
@@ -51,7 +61,8 @@ export const modifyMapResource = async (context: $TSContext, parameters: MapPara
...
```
Comment (jhockett) [inline packages/amplify-category-geo/src/service-utils/mapUtils.ts:65]: Same suggestion here, but the surrounding code looks so similar I wonder if it should be pulled into a function

```suggestion
  stateManager.setResourceInputsJson(
    undefined,
    category,
    parameters.name,
    { groupPermissions: parameters.groupPermissions },
  );
```

File: packages/amplify-category-geo/src/service-utils/placeIndexUtils.ts [hunk 4]
```diff
@@ -29,7 +35,8 @@ export const createPlaceIndexResource = async (context: $TSContext, parameters:
   const placeIndexStack = new PlaceIndexStack(new App(), 'PlaceIndexStack', { ...parameters, ...templateMappings, authResourceName });
   generateTemplateFile(placeIndexStack, parameters.name);
   saveCFNParameters(parameters);
-  stateManager.setResourceInputsJson(pathManager.findProjectRoot(), category, parameters.name, { groupPermissions: parameters.groupPermissions });
+  stateManager.setResourceInputsJson(pathManager.findProjectRoot(), category,
+    parameters.name, { groupPermissions: parameters.groupPermissions });
```
Comment (jhockett) [inline packages/amplify-category-geo/src/service-utils/placeIndexUtils.ts:39]: Same comment

```suggestion
  stateManager.setResourceInputsJson(
    undefined,
    category,
    parameters.name,
    { groupPermissions: parameters.groupPermissions },
  );
```

File: packages/amplify-category-geo/src/service-utils/placeIndexUtils.ts [hunk 5]
```diff
@@ -52,7 +62,8 @@ export const modifyPlaceIndexResource = async (context: $TSContext, parameters:
   const placeIndexStack = new PlaceIndexStack(new App(), 'PlaceIndexStack', { ...parameters, ...templateMappings, authResourceName });
   generateTemplateFile(placeIndexStack, parameters.name);
   saveCFNParameters(parameters);
-  stateManager.setResourceInputsJson(pathManager.findProjectRoot(), category, parameters.name, { groupPermissions: parameters.groupPermissions });
+  stateManager.setResourceInputsJson(pathManager.findProjectRoot(),
+    category, parameters.name, { groupPermissions: parameters.groupPermissions });
```
Comment (jhockett) [inline packages/amplify-category-geo/src/service-utils/placeIndexUtils.ts:66]: There should be a function to take care of saving state. Most of this code is duplicated 5 times

**Review by sobolk — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**